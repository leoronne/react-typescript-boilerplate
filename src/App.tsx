import Providers, { ThemeProvider } from './providers';
import Routes from './routes';

function App() {
  return (
    <Providers with={[ThemeProvider]}>
      <Routes />
    </Providers>
  );
}

export default App;
