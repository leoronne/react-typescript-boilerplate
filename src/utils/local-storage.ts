import { APP_SLUG } from 'shared/consts';

export const appLocalStorageSlug = `@${APP_SLUG}/`;

export const getItem = (key: string) => {
  try {
    return localStorage.getItem(`${appLocalStorageSlug}${key}`);
  } catch (err: any) {
    // eslint-disable-next-line no-console
    console.error(err?.message);
    return null;
  }
};

export const setItem = (key: string, value: string) => {
  try {
    localStorage.setItem(`${appLocalStorageSlug}${key}`, value);
  } catch (err: any) {
    // eslint-disable-next-line no-console
    console.error(err?.message);
  }
};
