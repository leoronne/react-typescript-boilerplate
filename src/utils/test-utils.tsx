import {
  screen,
  render as renderTestComponent,
  ByRoleMatcher,
  ByRoleOptions,
} from '@testing-library/react';

import Providers, { ThemeProvider } from 'providers';

export function render(Component: JSX.Element) {
  function App() {
    return <Providers with={[ThemeProvider]}>{Component}</Providers>;
  }

  return renderTestComponent(<App />);
}

export function getByRole(
  role: ByRoleMatcher,
  name?:
    | RegExp
    | string
    | ((accessibleName: string, element: Element) => boolean),
  options?: ByRoleOptions | undefined
): HTMLElement {
  return screen.getByRole(role, {
    name,
    ...options,
  });
}

export function getButton(
  name?:
    | RegExp
    | string
    | ((accessibleName: string, element: Element) => boolean)
): HTMLElement {
  return screen.getByRole('button', {
    name,
  });
}

export function queryButton(
  name?:
    | RegExp
    | string
    | ((accessibleName: string, element: Element) => boolean)
): HTMLElement | null {
  return screen.queryByRole('button', {
    name,
  });
}

export function getLink(
  name?:
    | RegExp
    | string
    | ((accessibleName: string, element: Element) => boolean)
): HTMLElement {
  return screen.getByRole('link', {
    name,
  });
}

export function getHeading(
  name?:
    | RegExp
    | string
    | ((accessibleName: string, element: Element) => boolean)
): HTMLElement {
  return screen.getByRole('heading', {
    name,
  });
}

export function getImage(
  name?:
    | RegExp
    | string
    | ((accessibleName: string, element: Element) => boolean)
): HTMLElement {
  return screen.getByRole('img', {
    name,
  });
}

export function queryHeading(
  name?:
    | RegExp
    | string
    | ((accessibleName: string, element: Element) => boolean)
): HTMLElement | null {
  return screen.queryByRole('heading', {
    name,
  });
}
