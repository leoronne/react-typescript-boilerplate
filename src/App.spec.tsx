import { render } from 'utils/test-utils';
import App from './App';

describe('<App />', () => {
  it('Should render', async () => {
    const app = render(<App />);

    expect(app).toMatchSnapshot();
  });
});
