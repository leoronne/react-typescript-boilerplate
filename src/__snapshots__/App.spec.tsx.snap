// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-aic56o"
      >
        <header
          class="css-1mh5dai"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1vivydv-MuiTypography-root-MuiLink-root"
            href="/"
          >
            <svg
              class="css-1wehgf4"
            >
              logo.svg
            </svg>
          </a>
          <button
            aria-label="Open settings menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-by6ct-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SettingsRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.5 12c0-.23-.01-.45-.03-.68l1.86-1.41c.4-.3.51-.86.26-1.3l-1.87-3.23c-.25-.44-.79-.62-1.25-.42l-2.15.91c-.37-.26-.76-.49-1.17-.68l-.29-2.31c-.06-.5-.49-.88-.99-.88h-3.73c-.51 0-.94.38-1 .88l-.29 2.31c-.41.19-.8.42-1.17.68l-2.15-.91c-.46-.2-1-.02-1.25.42L2.41 8.62c-.25.44-.14.99.26 1.3l1.86 1.41c-.02.22-.03.44-.03.67s.01.45.03.68l-1.86 1.41c-.4.3-.51.86-.26 1.3l1.87 3.23c.25.44.79.62 1.25.42l2.15-.91c.37.26.76.49 1.17.68l.29 2.31c.06.5.49.88.99.88h3.73c.5 0 .93-.38.99-.88l.29-2.31c.41-.19.8-.42 1.17-.68l2.15.91c.46.2 1 .02 1.25-.42l1.87-3.23c.25-.44.14-.99-.26-1.3l-1.86-1.41c.03-.23.04-.45.04-.68zm-7.46 3.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="PrivateSwipeArea-root css-x15wq9"
            style="width: 20px;"
          />
        </header>
        <div
          class="MuiBox-root css-1qb4zv3"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-9z7mrc-MuiTypography-root"
          >
            React + TypeScript Boilerplate
          </h1>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-175dzw9-MuiTypography-root"
          >
            This a simple front-end boilerplate for personal projects
          </h5>
        </div>
        <footer
          class="css-4wkrq6"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ede0sf-MuiTypography-root"
          >
            &lt;ronne.dev /&gt; © 2022
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-aic56o"
    >
      <header
        class="css-1mh5dai"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1vivydv-MuiTypography-root-MuiLink-root"
          href="/"
        >
          <svg
            class="css-1wehgf4"
          >
            logo.svg
          </svg>
        </a>
        <button
          aria-label="Open settings menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-by6ct-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SettingsRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.5 12c0-.23-.01-.45-.03-.68l1.86-1.41c.4-.3.51-.86.26-1.3l-1.87-3.23c-.25-.44-.79-.62-1.25-.42l-2.15.91c-.37-.26-.76-.49-1.17-.68l-.29-2.31c-.06-.5-.49-.88-.99-.88h-3.73c-.51 0-.94.38-1 .88l-.29 2.31c-.41.19-.8.42-1.17.68l-2.15-.91c-.46-.2-1-.02-1.25.42L2.41 8.62c-.25.44-.14.99.26 1.3l1.86 1.41c-.02.22-.03.44-.03.67s.01.45.03.68l-1.86 1.41c-.4.3-.51.86-.26 1.3l1.87 3.23c.25.44.79.62 1.25.42l2.15-.91c.37.26.76.49 1.17.68l.29 2.31c.06.5.49.88.99.88h3.73c.5 0 .93-.38.99-.88l.29-2.31c.41-.19.8-.42 1.17-.68l2.15.91c.46.2 1 .02 1.25-.42l1.87-3.23c.25-.44.14-.99-.26-1.3l-1.86-1.41c.03-.23.04-.45.04-.68zm-7.46 3.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="PrivateSwipeArea-root css-x15wq9"
          style="width: 20px;"
        />
      </header>
      <div
        class="MuiBox-root css-1qb4zv3"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-9z7mrc-MuiTypography-root"
        >
          React + TypeScript Boilerplate
        </h1>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-175dzw9-MuiTypography-root"
        >
          This a simple front-end boilerplate for personal projects
        </h5>
      </div>
      <footer
        class="css-4wkrq6"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ede0sf-MuiTypography-root"
        >
          &lt;ronne.dev /&gt; © 2022
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
